### ISSUE TITLE
Bundle different modules into a single chunk

### ISSUE THREAD

**Issue by luisherranz** [zeit/next.js#2480]
- [x] I have searched the [issues](https://github.com/zeit/next.js/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior

Users should be able to bundle different modules into one final chunk.

For example, users importing component `hello1.js` and `hello2.js` may want to import both modules at the same time:
```javascript
const Hello1 = dynamic(import('../components/hello1'))
const Hello2 = dynamic(import('../components/hello2'))
// This will request both hello1.js and hello2.js.
```

## Current Behavior

Webpack 3 allows you to do so using magic comments:
```javascript
const Hello1 = dynamic(/* webpackChunkName: "hello" */ import('../components/hello1'))
const Hello2 = dynamic(/* webpackChunkName: "hello" */ import('../components/hello2'))
// This should request only one hello.js with both modules.
```
But magic comments don't work on **nextjs**.

## Context

Sometimes a user requesting a module is going to need more modules in the future and you can bundle those in a single chunk to avoid server round trips.

## Your Environment

* Next.js version: 3.0.1-beta.5
* Environment name and version (e.g. Chrome 39, Node.js 5.4): Chrome 61, Node 6.11
* Operating System (Linux, maxOS, Windows): macOS

**Comment by arunoda** [comment]
Yes. This is a good feature. This doesn't work with Next.js because we have a our own implementation for dynamic imports.
(On top of webpack's `require.ensure`. That's for SSR purpose)

But I'd like to take this in near future.

**Comment by luisherranz** [comment]
Thanks @arunoda!

Is there a reason for not using Webpack's import? Or at least to merge it in the near future so we can have some missing Webpack features back, like magic comments or dynamic requires like ```dynamic(import(`../components/${someVar}`));```?

**Comment by arunoda** [comment]
I don't think we'll be able to use Webpack's import implementation. That's because our implementation is optimized for SSR as well.
Anyway, we only use Babel plugin and most of the heavy work is still done by webpack.

I don't think dynamic variable support will be something you can see in Next.js or Webpack ever. If we do so, then it's impossible to statically analyze imports.

**Comment by luisherranz** [comment]
Thanks for the explanation @arunoda.

**Comment by luisherranz** [comment]
Can react-universal-component (and webpack-flush and babel-plugin-universal-import) be used within next?

**Comment by timneutkens** [comment]
#4639 implements react-loadable, which is similar to react-universal-component, and allows for webpack's "magic comment" approach.
